### PR TITLE
Column state reducer helper tests

### PR DESCRIPTION
**Pull request by wcjordan**
Adding some tests for the column reducer state helper before I refactor it to remove the React components / dependency.

## Description

Adding tests.  They're pretty ugly / rough and don't cover all the edge cases.  I'll flesh them out further after I've refactored the state.

@KamranAsif any advice on making these tests easier to read would be appreciated.

## Motivation and Context

Hopefully prevents any regressions during refactoring

## How Has This Been Tested?

Running the tests.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.

### CODE REVIEW

File: src/ReactTouchHandler-test.js [hunk 1]
```diff
@@ -13,18 +13,18 @@ describe('ReactTouchHandler', function() {
 
   before(function () {
     clock = sinon.useFakeTimers();
+    sandbox = sinon.sandbox.create();
```
Comment (wcjordan) [inline src/ReactTouchHandler-test.js]: I think we only want to create the sandbox once

File: src/reducers/columnStateHelper-test.js [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+          fixed: false,
+          width: 200
+        })
+      ];
+
+      // --- Run Test ---
+      var result = columnStateHelper.initialize(oldState, {
+        children: children,
+        width: 200,
+        isColumnResizing: true
```
Comment (wcjordan) [inline src/reducers/columnStateHelper-test.js]: This is weird... I noticed if you update props while resizing and don't pass isColumnResizing in as a prop, we stop your resizing.  Is this behavior we want to keep or should be change it?  Are there any common cases where an HOC might change the props while someone's resizing a column?

File: src/reducers/columnStateHelper.js [hunk 3]
```diff
@@ -323,8 +323,8 @@ function reorderColumn(oldState, reorderData) {
       columnKey: columnKey,
       columnWidth: width,
       originalLeft: left,
-      columnsBefore: [],
```
Comment (wcjordan) [inline src/reducers/columnStateHelper.js]: these were unused, I think a rename was missed in a refactor at some point

**Comment by wcjordan** [comment]
I renamed FixedDataTableRoot-test but will rename back before committing.  I just wanted to have tests passing.

File: src/reducers/columnStateHelper.js (hunk 3 above)
Comment (KamranAsif) [inline src/reducers/columnStateHelper.js]: Can we just remove?

File: src/reducers/columnStateHelper-test.js (hunk 2 above)
Comment (KamranAsif) [inline src/reducers/columnStateHelper-test.js]: This seems analogous to the other props, isColumnReordering and isColumnResizing. I'm not sure why they are there but we shouldn't break the API yet

**Comment by KamranAsif** [comment]
LGTM

File: src/reducers/columnStateHelper.js (hunk 3 above)
Comment (wcjordan) [inline src/reducers/columnStateHelper.js]: They're removed, but columnBefore and columnAfter (not columns) are new and needed

**Comment by wcjordan** [comment]
I'm having trouble with getting Travis-CI to run this, so I'm going to merge squash with the FixedDataTableRoot test rename.  Then I'll rename it back after I get a green (and break the branch again).
